File: Data Structures & Algorithms/baseball-game/submission-1.java
class Solution {
    public int calPoints(String[] operations) {
        Stack<Integer> stack = new Stack<>();
        int total = 0;

        // ["5","-2","4","C","D","9","+","+"]
        // [5,-2,4,9]
        // total = 32
        for(int i=0; i<operations.length; i++){
            String operation = operations[i];

            if(operation.equals("+")){
                int first = stack.peek();
                int second = stack.get(stack.size() - 2);
                int sum = first + second;
                stack.add(sum);
                total += sum;
            } else if(operation.equals("C")){
                total -= stack.pop();
            } else if(operation.equals("D")){
                int num = stack.peek() * 2;
                stack.add(num);
                total += num;
            } else{
                stack.add(Integer.parseInt(operation));
                total += Integer.parseInt(operation);
            }
        }
        return total;
    }
}

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-1.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []

        for elem in tokens:
            if elem in "+-*/":
                y = int(stack.pop())
                x = int(stack.pop())

                total = self.calc(x, y, elem)
                stack.append(str(total))
                continue
            stack.append(elem)
        return int(stack[0])

    def calc(self, x: int, y: int, op: str):
        if op == "+":
            return x + y
        elif op == "-":
            return x - y
        elif op == "/":
            return int(x / y)
        
        return x * y